github-actions/rebase-pr/entrypoint.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+set -e
+
+# Set email and name config
+git config --global user.email "devops+rebase@angular.io"
+git config --global user.name "Angular Rebase Bot"
+
+# Exit for code a successful run
+SUCCESS_EXIT_CODE=0
+
+# The repository name and branch name for the PRs head and base
+HEAD_REPOSITORY=$(jq -r ".pull_request.head.repo.full_name" "$GITHUB_EVENT_PATH")
+HEAD_BRANCH=$(jq -r ".pull_request.head.ref" "$GITHUB_EVENT_PATH")
+BASE_REPOSITORY=$(jq -r ".pull_request.base.repo.full_name" "$GITHUB_EVENT_PATH")
+BASE_BRANCH=$(jq -r ".pull_request.base.ref" "$GITHUB_EVENT_PATH")
+
+# Change to the github workspace and initialize the repo if needed.
+cd "$GITHUB_WORKSPACE"
+git init
+
+# Add head and base repos as remotes, then fetch the remote branches
+git remote add head https://x-access-token:$TOKEN@github.com/$HEAD_REPOSITORY
+git remote add base https://x-access-token:$TOKEN@github.com/$BASE_REPOSITORY
+git fetch head "$HEAD_BRANCH";
+git fetch base "$BASE_BRANCH";
+
+# Checkout the PR change branch from its remote
+git checkout --track head/"$HEAD_BRANCH"
+
+# Perform rebase of PR
+git rebase base/"$BASE_BRANCH";
+
+# If the rebase attempt fails, exit with a failing (1) exit code.
+if [ $? -ne 0 ]; then
+  # TODO(josephperrott): Push a comment to the PR asking the author to rebase.
+  exit 1;
+fi
+
+# Push the rebased code back to the remote, using force as this is required
+# for rebases, then exit with a success exit code.
+git push --force-with-lease head "$HEAD_BRANCH";
+exit $SUCCESS_EXIT_CODE;
```

Comment:
This feels refundant as well :thinking_face:

Reply:
The reason I have it there is because while `0` is the current success code.  I was unsure if this will always be the case for Github Actions as they have mentioned in the Actions issues repo that they may have other result codes as well.